torch/lib/c10d/CMakeLists.txt

```diff
@@ -127,12 +119,14 @@ target_compile_options(c10d PUBLIC
   -Wno-unknown-pragmas
   )
 
-# c10d links to Caffe2/ATen, but the targets don't add TH/THC to the include path
-target_include_directories(c10d PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../tmp_install/include/TH)
 
-if(C10D_USE_CUDA)
-  target_include_directories(c10d PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../tmp_install/include/THC)
-endif()
+add_dependencies(c10d caffe2_library)
```

Comment:
same comments as for THD, depend on `caffe2`, and remove the manual include_directories section below